### PR TITLE
Add support for Django 1.10-style middleware

### PR DESCRIPTION
**Pull request by davegaeddert**
There may be more to do here (or you may want to handle it differently), but this seems to be an easy way to add support for the new way that middleware works in Django 1.10.

https://docs.djangoproject.com/en/1.10/topics/http/middleware/#upgrading-middleware

### CODE REVIEW

**Comment by Miserlou** [comment]
I'm into this, but the test is failing. Can you take a look at that before merge?

**Comment by davegaeddert** [comment]
I can try and take a look, looks like the same test is failing on master though (and then also of the other PR's). https://travis-ci.org/Miserlou/django-easy-timezones/jobs/173703797 I'm guessing it has something to do with using strings to reference the views instead of using the view classes themselves, which was maybe removed in Django 1.8? https://docs.djangoproject.com/en/1.7/topics/http/urls/#passing-strings-instead-of-callable-objects

I also have to admit that now I'm a little unsure about my addition -- I use django quite a bit but have never published a package that can be used with different versions of django... I'm kinda assuming the change I made can cause issues in <1.10 if that mixin class doesn't exist. Any thoughts on that? Seems like maybe the existing issue in the tests is a similar scenario. Might be worth looking into tox to test against multiple django versions, and adding some logic in the code to check which django version it is running under?

Sorry I think I raised more questions than answers! Let me know what you think the best route forward is. I'd be happy to dig into a little bit of this if you want me to.

File: easy_timezones/middleware.py [hunk 1]
```diff
@@ -84,7 +93,7 @@ def process_request(self, request):
         if tz:
             timezone.activate(tz)
             request.session['django_timezone'] = str(tz)
-            if getattr(settings, 'AUTH_USER_MODEL', None):
+            if getattr(settings, 'AUTH_USER_MODEL', None) and getattr(request, 'user', None):
```
Comment (davegaeddert) [inline easy_timezones/middleware.py]: Django 1.7 fails without this (and without specifying `AUTH_USER_MODEL` in settings)...

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
-coverage==4.1
```
Comment (davegaeddert) [inline requirements.txt]: Felt like maybe this didn't need to be in the requirements that get imported into setup.py? Test settings got separated anyway to remove the django requirement so tox could take care of that (gives duplicate error if in both).

**Comment by davegaeddert** [comment]
Hey @Miserlou, I had a few minutes to spend on this tonight. I went ahead and added tox so that Django 1.7, 1.8, 1.9, and 1.10 all got tested against, with different python versions. It pointed out a handful of failures (including the one you currently have in master). It also answered my question about the code I added -- which also failed <1.10 and now conditionally uses that MiddlewareMixin based on the Django version. I tried to keep the rest of the code change as minimal as possible, but did go ahead and fix the test errors that I was seeing.

Let me know what you think.

**Comment by Miserlou** [comment]
Fantastic!

**Comment by davegaeddert** [comment]
Great! Last question -- any idea if/when this will hit pypi?

**Comment by Miserlou** [comment]
Already has! :) https://pypi.python.org/pypi/django-easy-timezones/

**Pull request merged**